client/client-multi/client-c/src/kaa/platform-impl/posix/file_utils.c

```diff
@@ -17,18 +17,24 @@
 #include <platform/file_utils.h>
 #include <stdint.h>
 #include <platform/stdio.h>
+#include <platform/kaa_client_properties.h>
 #include "utilities/kaa_mem.h"
 #include "kaa_common.h"
 
+char *kaa_binary_file_path;
 
 int posix_binary_file_read(const char *file_name, char **buffer, size_t *buffer_size, bool *needs_deallocation)
 {
-    KAA_RETURN_IF_NIL4(file_name, buffer, buffer_size, needs_deallocation, -1);
+    kaa_binary_file_path = KAA_MALLOC(sizeof(kaa_client_props_get()) + sizeof(file_name));
+    sprintf(kaa_binary_file_path,"%s%s",kaa_client_props_get(),file_name);
+
+    KAA_RETURN_IF_NIL4(kaa_binary_file_path, buffer, buffer_size, needs_deallocation, -1);
     *buffer = NULL;
     *buffer_size = 0;
     *needs_deallocation = true;
 
-    FILE* file = fopen(file_name, "rb");
+    FILE* file = fopen(kaa_binary_file_path, "rb");
+    KAA_FREE(kaa_binary_file_path);
     KAA_RETURN_IF_NIL(file, -1);
```

Comment:
Don't use this macro. Free dynamically allocated memory upon returning from the function